File: InterviewCampProblems/InterviewCampProblems.Test/UnitTest6.cs
/*
Given a sorted array in non-decreasing order, return an array of squares of each number, also in non-decreasing order. For example:[-4,-2,-1,0,3,5] -> [0,1,4,9,16,25]
*/

using System;
using InterviewCampProblems.Code;
using Xunit;

namespace InterviewCampProblems.Test {
    public class UnitTest6 {
        [Fact]
        public void Test1 () {

            int[] expected = new int[] { 0, 1, 4, 9, 16, 25 };
            int[] arr = new int[] {-4, -2, -1, 0, 3, 5 };
            var actual = Class6.SquareOfEachNumber (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test2 () {

            int[] expected = new int[] { };
            int[] arr = new int[] { };
            var actual = Class6.SquareOfEachNumber (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test3 () {

            int[] expected = null;
            int[] arr = null;
            var actual = Class6.SquareOfEachNumber (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test4 () {

            int[] expected = new int[] { 1, 4 };
            int[] arr = new int[] {-2, 1 };
            var actual = Class6.SquareOfEachNumber (arr);
            Assert.Equal (expected, actual);
        }
    }
}

File: Java/InterviewCamp/src/test/java/UnitTest6.java
//Given a sorted array in non-decreasing order, return an array of squares of each number, also in non-decreasing order. For example:[-4,-2,-1,0,3,5] -> [0,1,4,9,16,25]

import org.junit.Assert;
import org.junit.Test;

public class UnitTest6 {

    @Test
    public void When_Input_Correct() {

        //Arrange
        int[] arr = new int[] {-4,-2,-1,0,3,5};
        int[] outputArr = new int[] {0,1,4,9,16,25};

        //Act
        int[] output = Class6.squareArrayElement(arr);

        //Assert
        Assert.assertArrayEquals(output,outputArr);

    }
}